app/src/main/java/org/wikipedia/page/PageActivity.java

```diff
@@ -683,6 +684,11 @@ public void onActivityResult(int requestCode, int resultCode, final Intent data)
             } else if (resultCode == TabActivity.RESULT_LOAD_FROM_BACKSTACK) {
                 pageFragment.reloadFromBackstack();
             }
+        } else if (requestCode == Constants.ACTIVITY_REQUEST_IMAGE_CAPTION_EDIT
+                && resultCode == RESULT_OK) {
+            pageFragment.refreshPage();
+            FeedbackUtil.showMessage(this, getString(R.string.description_edit_success_saved_image_caption_in_lang_snackbar,
+                    app.language().getAppLanguageLocalizedName(StringUtils.defaultString(pageFragment.getLeadingImageEditLang()))));
```

Comment:
Since the return value of `getLeadingImageEditLang()` can be null/empty, it won't make sense to feed it into a string that always expects a language parameter. I suggest checking if the language is empty, and using the non-language-specific string `description_edit_success_saved_image_caption_snackbar` in that case.